### PR TITLE
Snapshot support for static PVC

### PR DESCRIPTION
**Pull request by smitaraut**
Handling of PVC creation from a snapshot that was taken for a statically created PVC.

generate_pv script has changed to accommodate fileset parameter. If you need snapshot to be created from a static volume, you need to create static volume using independent fileset is the volume handle.

Doc update will be needed for thus

### CODE REVIEW

File: tools/generate_pv_yaml.sh [hunk 1]
```diff
@@ -29,6 +30,7 @@ fullUsage(){
 echo "Usage: $0
 		-f|--filesystem <Name of Volume's Source Filesystem>
 		-l|--linkpath <full Path of Volume in Primary Filesystem>
+    -F|--fileset <name of source fileset>
```
Comment (deeghuge) [inline tools/generate_pv_yaml.sh:33]: Hi Smita, I see some spacing issue in this file. please take care of the same.

File: driver/csiplugin/controllerserver.go [hunk 2]
```diff
@@ -693,15 +692,22 @@ func (cs *ScaleControllerServer) GetVolIdMembers(vId string) (scaleVolId, error)
...
 		vIdMem.ClusterId = splitVid[0]
 		vIdMem.FsUUID = splitVid[1]
 		SlnkPart := splitVid[2]
 		slnkSplit := strings.Split(SlnkPart, "=")
 		if len(slnkSplit) < 2 {
 			return scaleVolId{}, status.Error(codes.Internal, fmt.Sprintf("Invalid Volume Id : [%v]", vId))
 		}
-		vIdMem.SymLnkPath = slnkSplit[1]
-		vIdMem.IsFilesetBased = false
+		if strings.EqualFold(slnkSplit[0], "fileset") {
```
Comment (deeghuge) [inline driver/csiplugin/controllerserver.go:704]: Hi Smita, Sorry I don't understand this part of change ? Can you please explain

Comment (madhuthorat) [reply driver/csiplugin/controllerserver.go:704]: > Hi Smita, Sorry I don't understand this part of change ? Can you please explain why it is required

I think the (case insensitive) check is there to identify whether after filesystem UUID we have next argument as "fileset=..." or "path=..." and based on that set further variables.

**Review by madhuthorat — APPROVED** [review]
Looks good to me. Approved (except Deepak's comment for spacing issue which needs to be handled).

File: driver/csiplugin/controllerserver.go (hunk 2 above)
Comment (deeghuge) [reply driver/csiplugin/controllerserver.go:704]: but this if is for LW volumes if len(splitVid) == 3 so there is never going to be case of fileset based volume. fileset volume handle splitting is taken care in next if where handle is of divided into 4 part.

Comment (smitaraut) [reply driver/csiplugin/controllerserver.go]: Right Deepak, this change was introduced as for fileset static provisioning the format was going to be **<cluster_id>;<filesystem_uuid>;fileset=<fileset_id>** without the "path". But it was later changed to also include path, hence reverting this change.

File: tools/generate_pv_yaml.sh (hunk 1 above)
Comment (smitaraut) [reply tools/generate_pv_yaml.sh]: Corrected this

**Review by deeghuge — APPROVED** [review]
lgtm

**Review by Jainbrt — APPROVED** [review]
LGTM

**Pull request merged**